app/assets/javascripts/discourse/app/components/sidebar/api_panels.js

```diff
@@ -0,0 +1,19 @@
+import Component from "@glimmer/component";
+import { getOwner, setOwner } from "@ember/application";
+import { inject as service } from "@ember/service";
+import { cached } from "@glimmer/tracking";
+
+export default class SidebarApiPanels extends Component {
+  @service siteSettings;
+  @service currentUser;
+  @service site;
+
+  @cached
+  get customSections() {
+    return this.args.panel.sections.map((customSection) => {
+      const section = new customSection({ sidebar: this });
```

Comment:
We need to avoid re initialising an object in a getter each time it is called as it can lead to performance problems. Is there a way we can avoid doing this?